Libraries/Opc.Ua.Server/Diagnostics/MonitoredNode.cs

```diff
@@ -253,7 +253,22 @@ public void OnReportEvent(ISystemContext context, NodeState node, IFilterTarget
                 lock (NodeManager.Lock)
```

Comment:
wouldn´t adding this code in line 252 keep the lock contention lower and achieve the same effect and be more in line with the syntax of the checks before:

```c#
// skip event reporting if session does not match monitored items session
if (context?.SessionId != monitoredItem?.Session.Id)
 {
        continue;
}
```

Reply:
Hi @romanett,

First of all, thank you for your suggestion.
I chose not to skip the event reporting in case of a missing session for two reasons:

1. To ensure backward compatibility.
2. To provide a way to broadcast an event to all connected clients without looping through all active sessions. This is useful when an event is generated on the server side and needs to be reported to each session.

I remain open to any other suggestions or comments you may have.